### PR TITLE
Feat: add additional meta information to resourcec payload

### PR DESCRIPTION
**Pull request by omarsilva1**
# Pull Request
This PR adds additional meta information to the payload when publishing a resource.

## How Has This Been Tested?
I only tested this locally with postman. It would be good to test this with the UI @richardtreier 
**Update**: yesterday I tested it with the UI and everything seems to work fine.

**Test Configuration**:
I used the custom intellij run configuration.

## Linked Issue(s)
- closes https://github.com/sovity/edc-extensions/issues/7

# Checklist
- [x] I have **formatted the title** correctly and precisely
- [x] My code follows the **style guidelines** of this project
- [x] I have performed a **self-review** of my own code
- [x] I have **commented** my code, particularly in hard-to-understand areas and public classes/methods
- [x] I have made corresponding changes to the **documentation**
- [x] My changes generate **no new warnings** (performed checkstyle check locally)
- [x] I have added **tests that prove my fix** is effective or that my feature works
- [x] New and existing unit **tests pass locally** with my changes
- [x] Any dependent **changes have been merged** and published in downstream modules
- [x] I have added/updated **copyright headers**

### CODE REVIEW

File: extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java [hunk 1]
```diff
@@ -46,42 +53,159 @@ public RegisterResourceRequestSender(ObjectMapper objectMapper) {
...
+
+    private ObjectNode getTransportModeJson(String transportMode) {
+        return getPropertyJson(transportMode);
+    }
+
+    private ObjectNode getDataCategoryJson(String dataCategory) {
+        return getPropertyJson(dataCategory);
+    }
+
+    private ObjectNode getPropertyJson(String property) {
```
Comment (richardtreier) [inline extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java:135]: better name: buildStringProperty

**Review by richardtreier — CHANGES_REQUESTED** [review]

File: extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java [hunk 2]
```diff
@@ -46,42 +53,159 @@ public RegisterResourceRequestSender(ObjectMapper objectMapper) {
...
+                        ._mediaType_(new IANAMediaTypeBuilder()._filenameExtension_(mediaType).build())
+                        .build())
+                ._resourceEndpoint_(List.of(new ConnectorEndpointBuilder()
+                        ._endpointDocumentation_(URI.create(endpointDocumentation))
+                        ._accessURL_(registerResourceMessage.affectedResourceUri())
+                        .build()))
                 .build();
-        return objectMapper.writeValueAsString(resource);
+
+        ObjectNode json = (ObjectNode) objectMapper.readTree(objectMapper.writeValueAsString(resource));
```
Comment (efiege) [inline extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java:106]: var

**Review by efiege — CHANGES_REQUESTED** [review]

File: extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java [hunk 3]
```diff
@@ -46,42 +53,159 @@ public RegisterResourceRequestSender(ObjectMapper objectMapper) {
...
+    private ObjectNode getTransportModeJson(String transportMode) {
+        return getPropertyJson(transportMode);
+    }
+
+    private ObjectNode getDataCategoryJson(String dataCategory) {
+        return getPropertyJson(dataCategory);
+    }
+
+    private ObjectNode getPropertyJson(String property) {
+        ObjectNode transportModeJson = objectMapper.createObjectNode();
```
Comment (tmberthold) [inline extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java:136]: Maybe _var_ and why is the variable called `transportModeJson`?

File: extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java (hunk 1 above)
Comment (omarsilva1) [reply extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java]: done

File: extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java (hunk 3 above)
Comment (omarsilva1) [reply extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java:136]: Good point, I forgot to change it when refactoring the method. It's called "json" now.

File: extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java (hunk 2 above)
Comment (omarsilva1) [reply extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java]: done

**Review by efiege — APPROVED** [review]

**Pull request merged**